java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/DateConsumer.java

```diff
@@ -123,9 +105,10 @@ public NonNullableDateConsumer(DateDayVector vector, int index, Calendar calenda
     public void consume(ResultSet resultSet) throws SQLException {
       Date date = calendar == null ? resultSet.getDate(columnIndexInResultSet) :
           resultSet.getDate(columnIndexInResultSet, calendar);
-      int day = (int) TimeUnit.MILLISECONDS.toDays(date.getTime());
-      if (day < 0 || day > MAX_DAY) {
-        throw new IllegalArgumentException("Day overflow: " + day);
+      long dayLong = TimeUnit.MILLISECONDS.toDays(date.getTime());
+      int day = (int) dayLong;
+      if (day != dayLong) {
+        throw new IllegalArgumentException("Day overflow: " + dayLong);
```

Comment:
Maybe we should also use Math.toIntExact here?

Reply:
Ah yeah my fault, added.